src/compiler/semver.ts

```diff
@@ -319,8 +324,9 @@ function parseHyphen(left: string, right: string, comparators: Comparator[]) {
     if (!isWildcard(rightResult.major)) {
         comparators.push(
             isWildcard(rightResult.minor) ? createComparator("<", rightResult.version.increment("major")) :
-            isWildcard(rightResult.patch) ? createComparator("<", rightResult.version.increment("minor")) :
-            createComparator("<=", rightResult.version));
+                isWildcard(rightResult.patch) ? createComparator("<", rightResult.version.increment("minor")) :

```

Comment:
this is unfortunate; line 327 was wrong and 328 right. Now both 327 and 328 are the same (better), but both wrong.
Still an improvement.

Reply:
Old thread, but I'm not sure that this is actually inconsistent?

[Playground link](https://dprint.dev/playground/#code/JYMwBAFAhMDODqwA2ATAxgQwE4ol4A5gBYAuASgKawCuSJAdALYYBWA9lgJSdgDeAsACgwIsGjaMADtgwkOsepOqwiEIaI1g4iVJhx5CpSjTpNgAOw48A-GKwVZFAMITpWWRwgAiADxeANGD4xORUtAwAbhRYsMBs5vQWaPaMFOYk3szsWF7cYABc6prF2sjo2LjBRmGm0iRoRDZ2DiTOrjJyWN5+gVWhJpHRsfGJ5skUqemZFhy5PIXCxZrjji5SHZ6+ALwBQYb94fRRMXHmnP5FIpwA3EIAvkJC4uvunQpKKmqLIqW6FQYhYyHRgzLhgWwrVprNweLq+XZ9IGmY7DBJJFJpDJeLKzPILJZaBBlPSVfZIhh1BpNSFtF6w7oIsk1QYnEboiaY6aWHJ4y7LeyrdqvTY+Ha9JkDI5DU7nIQ3R6CcTmWAkMAADzAW0JOnK+kRzLM3OpAqhQvp8PFgINKNOo3GkyxOJ58z5Gl+utJVsllMa4OagrpnQZluqkptbLGGKm2NBcwKrtENOhGzhosZXsO4bOt0ECtAkHdJIBoeBoONLVpMKDFr2GeR0oj9s52NYuJd300hf++u9sipfqTZurPVrJfrrLRkY50ZB3Lj+KWg8DIrFo4O49ReV4D1zghA1DGJFOYBAbDYEB4Ag79hI1Cw5m1xO7EtLRoHJsrKeDa-JUondqjR1W2deMOzdIk-j1F9aj7X0IQ-ZNhVTdMxxZVEAOnR1Y15MDEwQocVxQ9c0Jle4gA/config/N4WAUABF1QRAlgOwCYFNEBcDq9kYBawBcEALADTgywA2SqOehJAjAAweWTSyKoDuAGXoBpJMmIRYAQwCuGAPawu1WQGdUAFWkAjNZIBm0mhpUxqGgLbwAwgpoLE+kjJr9pAT31nzUgI6yChioAMoYHjSokrAADgBOqAaocQAiCrI6kcpUvv6BwQAKcQoxzlIAxo5q8GrBmNncvrDqqABCcdLlqGWw-PjoAHJBIUgA5pHCfA25UjodXQUK1RjwjtFq0paok6gAqoiRamoAEtKIo2PTubDV55GtCsgei8uriOub2-RXTXwAHhg7JhijQAGIOfgveArNYuf4YHY-cywDAdeB0c52SyWaQ9Rw0DwAWVkNBWiJ8yPiiWSp3OlxIRhMqAp1BKyWkijiUJh7xcOKQGGkSAaOSa0jixX4oNkiHKPIAdC0CuL0ARut1olSknEhlMWTxKihoW9jABRP5Uo5veUY1AFZLmy3VWEQRmmUXIgwyuVvR0JK2OeVqGKdVAAQQMwTi0tlPJEqA8-AUcQkJFRsmZHuo8EsMWTGBSqHKNHFHOtBmTXWJpPgiLTcQz+rghtqDblyc0HhiqCDIa6EajAwE8cTydTEHTmcayJb6blI1GiA5sgSvdDA+SQ-4I6TKckk6bUksjxJPbU+YAkrn84Xi6WeT1+Zghe9D7Bj8hT0H846b0WSx0D7RE+grCm+qAWn+d6Ada55xBgAyfMgv7wY+L6ga+WY8DmebwbeAFloGcEIUhV64RgaECi+sDgAAvuAQA/language/typescript)